contributor-docs/adrs/adr-020-live-regions.md

```diff
@@ -0,0 +1,233 @@
+However, both of these approaches do not announce consistently across screen
+readers. This could be due to live regions being injected dynamically into the document (this includes loading content into the document via React), dynamically changing the visibility of a live region, or some other technique causing an announcement to not be announced.
+
+For more information about the ways in which live regions may not work as
+expected, visit: [Why are my live regions not working?](https://tetralogical.com/blog/2024/05/01/why-are-my-live-regions-not-working/)
+
+### Links & Resources for ARIA Live regions
+
+- https://www.sarasoueidan.com/blog/accessible-notifications-with-aria-live-regions-part-1/
+- https://www.scottohara.me/blog/2022/02/05/are-we-live.html
+
+## Decision
+
+In order to have a common interop point for live region announcements, Primer React will
+make use of a `live-region` custom element from `@primer/live-region-element`.
+This package will be included and published from the `primer/react` repo.
+
+The custom element exposes a way to make announcements that can be used
+across frameworks. This makes it useful not only for Primer but GitHub as a
+whole. The `@primer/live-region-element` exports two helpers to use for making
+announcements: `announce()` and `announceFromElement`. Both helpers can be used
+when working in Primer and by teams at GitHub.
+
+In addition, `@primer/react` will leverage and export the following helpers for
+use within Primer React and GitHub:
+
+- The `AriaStatus` component to correspond with `role="status"`
+- The `AriaAlert` component to correspond with `role="alert"`
+
+Within `@primer/react`, we should lint against usage of `aria-live` and the
+corresponding roles (if possible) and suggest using these alternatives instead.
+
+> [!NOTE]
+> Both `AriaStatus` and `AriaAlert` will trigger an announcement when the component is
+> rendered. As a result, they should only be used for dynamically rendered
+> content. Otherwise, they will trigger announcements on page load. In cases
+> where they should always be present, then the first message passed to the
+> component should be an empty string. Changes to the content of the component
+> will trigger subsequent announcements.
+
+### Impact
+
+This decision will impact existing usage of `aria-live`, `role="status"`, and
+`role="alert"` for the components listed below. These components will need to be
+updated to use the new approach, using one of the following approaches:
+
+- Use `announce()` or `announceFromElement()`
+- Use `AriaStatus` or `AriaAlert`
+
+In addition, we should make sure that `<live-region>` is successfully included
+in GitHub.
+
+#### Instances of `aria-live="polite"`
+
+- InputValidation
+- SelectPanel
+- TreeView
+
+#### Instances of `aria-live="assertive"`
+
+- InlineAutocomplete
+
+### Instances of `role="status"`
+
+- LiveRegion
+- TreeView
+- ActionList examples
```

Comment:
I think Spinner would use `role="status"` too once [this PR](https://github.com/primer/react/pull/4140) is merged

```suggestion
- ActionList examples
- Spinner
```